pkg/solana/txm/txm.go

```diff
@@ -466,19 +476,17 @@ func (txm *Txm) processConfirmations(ctx context.Context, client client.ReaderWr
 				switch status.ConfirmationStatus {
 				case rpc.ConfirmationStatusProcessed:
 					// if signature is processed, keep polling for confirmed or finalized status
+					// we also need to check if a potential re-org has occurred for this sig and handle it
 					txm.handleProcessedSignatureStatus(sig)
-					continue
+					txm.handleReorg(ctx, sig, status)
```

Comment:
Thinking a little more about this, I think it might be better if we handle re-orgs before `handleProcessedSignatureStatus`. Otherwise, we would falsely log an error in `handleProcessedSignatureStatus` because the tx would not be found in the broadcasted/processed map if a re-org has occurred. If we handle re-orgs first, we would put the tx back into the broadcasted state to avoid the error log.

Reply:
done! lmk if that is what you were thinking for both cases